### PR TITLE
Changing format for api urls

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #7686

### CODE REVIEW

File: pulp_ansible/app/tasks/collections.py [hunk 1]
```diff
@@ -335,15 +335,39 @@ async def _fetch_collections(self):
...
+                root = f"{root}/api/"
+                downloader = remote.get_downloader(url=root)
+                api_data = parse_metadata(await downloader.run())
+
+            if "v3" in api_data.get("available_versions", {}):
+                api_version = 3
+            elif "v2" in api_data.get("available_versions", {}):
+                api_version = 2
+            else:
+                raise RuntimeError(_("Unsupported API versions at {}").format(root))
```
Comment (fao89) [inline pulp_ansible/app/tasks/collections.py:354]: This is great! I really like this solution!

**Review by fao89 — APPROVED** [review]
Thank you David!

File: CHANGES/7686.bugfix [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Supporting url formats that conform to ansible-galaxy cli (e.g. "https://galaxy.ansible.com" and
+"https://galaxy.ansible.com/api").
```
Comment (bmbouter) [inline CHANGES/7686.bugfix:2]: Do we want to allow slashes on the end here? It's kind of a silly thing, but the docs I'm reading have slashes. @daviddavis wdyt?

Comment (bmbouter) [reply CHANGES/7686.bugfix:2]: https://docs.ansible.com/ansible/latest/user_guide/collections_using.html#configuring-the-ansible-galaxy-client

Comment (daviddavis) [reply CHANGES/7686.bugfix:2]: Yup, I can do it.

Comment (daviddavis) [reply CHANGES/7686.bugfix:2]: [Done](https://github.com/pulp/pulp_ansible/pull/397/files#diff-d8871ad5c606deb92237ce777a88530a3522c2ae112096a0e105fb42f8337cf7R75).

File: pulp_ansible/tests/functional/constants.py [hunk 3]
```diff
@@ -84,36 +82,8 @@
 collections:
 - name: testing.ansible_testing_content
   version: ">=1.0.0,<=2.0.0"
-  source: https://galaxy-dev.ansible.com/api/v2/collections
+  source: https://galaxy-dev.ansible.com
 - testing.k8s_demo_collection
 """
 
-TOKEN_AUTH_COLLECTIONS_URL = "https://cloud.redhat.com/api/automation-hub/v3/collections/"
```
Comment (bmbouter) [inline pulp_ansible/tests/functional/constants.py:91]: ty for removing all of these

File: docs/workflows/collections.rst [hunk 4]
```diff
@@ -125,6 +125,7 @@ Remote GET Response::
         "pulp_created": "2019-04-29T13:51:10.860792Z",
         "pulp_href": "/pulp/api/v3/remotes/ansible/collection/e1c65074-3a4f-4f06-837e-75a9a90f2c31/",
         "pulp_last_updated": "2019-04-29T13:51:10.860805Z",
+        "requirements_file": "collections:\n  - testing.ansible_testing_content",
```
Comment (bmbouter) [inline docs/workflows/collections.rst:128]: I compiled these docs and then tested them and they worked for me.

File: pulp_ansible/tests/functional/api/collection/v2/test_sync.py [hunk 5]
```diff
@@ -61,7 +62,24 @@ def test_sync(self):
...
+        remote = self.client.post(ANSIBLE_COLLECTION_REMOTE_PATH, body)
+        self.addCleanup(self.client.delete, remote["pulp_href"])
+
+        # Sync the repository.
+        self.assertEqual(repo["latest_version_href"], f"{repo['pulp_href']}versions/0/")
+        sync(self.cfg, remote, repo)
+        repo = self.client.get(repo["pulp_href"])
+        self.assertEqual(repo["latest_version_href"], f"{repo['pulp_href']}versions/1/")
+
+    def test_sync_with_slash(self):
```
Comment (bmbouter) [inline pulp_ansible/tests/functional/api/collection/v2/test_sync.py:75]: Nice :+1:

**Review by bmbouter — APPROVED** [review]
I think this is a big improvement! This tested fully for me. Thank you!

**Pull request merged**